test/e2e/sshkey_test.go

```diff
@@ -0,0 +1,166 @@
+					Age().Newline().
+					End(),
+				out,
+			)
+		})
+
+		t.Run("json", func(t *testing.T) {
+			out, err := runCommand(t, "ssh-key", "list", "-o=json")
+			require.NoError(t, err)
+			assertjson.Equal(t, []byte(fmt.Sprintf(`
+[
+  {
+    "id": %d,
+    "name": %q,
+    "fingerprint": %q,
+    "public_key": %q,
+    "labels": {
+      "baz": "qux",
+      "foo": "bar"
+    },
+    "created": "<ignore-diff>"
+  }
+]`, sshKeyID, sshKeyName, fingerprint, pubKey)), []byte(out))
+		})
+	})
+
+	t.Run("remove-label", func(t *testing.T) {
+		out, err := runCommand(t, "ssh-key", "remove-label", strconv.FormatInt(sshKeyID, 10), "baz")
+		require.NoError(t, err)
+		assert.Equal(t, fmt.Sprintf("Label(s) baz removed from SSH Key %d\n", sshKeyID), out)
+	})
+
+	t.Run("update-name", func(t *testing.T) {
+		sshKeyName = withSuffix("new-test-ssh-key")
+		out, err := runCommand(t, "ssh-key", "update", strconv.FormatInt(sshKeyID, 10), "--name", sshKeyName)
+		require.NoError(t, err)
+		assert.Equal(t, fmt.Sprintf("SSHKey %d updated\n", sshKeyID), out)
+	})
+
+	t.Run("describe", func(t *testing.T) {
+		out, err := runCommand(t, "ssh-key", "describe", strconv.FormatInt(sshKeyID, 10))
+		require.NoError(t, err)
+		assert.Regexp(t, NewRegex().Start().
+			Lit("ID:").Whitespace().Lit(strconv.FormatInt(sshKeyID, 10)).Newline().
+			Lit("Name:").Whitespace().Lit(sshKeyName).Newline().
+			Lit("Created:").Whitespace().UnixDate().Lit(" (").HumanizeTime().Lit(")").Newline().
+			Lit("Fingerprint:").Whitespace().Lit(fingerprint).Newline().
+			Lit("Public Key:").Newline().Lit(pubKey).
+			Lit("Labels:").Newline().
+			Lit("  foo:").Whitespace().Lit("bar").Newline().
+			End(),
+			out,
+		)
+	})
+
+	t.Run("delete", func(t *testing.T) {
+		out, err := runCommand(t, "ssh-key", "delete", strconv.FormatInt(sshKeyID, 10))
+		require.NoError(t, err)
+		assert.Equal(t, fmt.Sprintf("SSH Key %d deleted\n", sshKeyID), out)
+	})
+}
+
+func createSSHKey(t *testing.T, name string, args ...string) (int64, error) {
+	t.Helper()
+	t.Cleanup(func() {
+		_, _ = client.SSHKey.Delete(context.Background(), &hcloud.SSHKey{Name: name})
+	})
+
+	out, err := runCommand(t, append([]string{"ssh-key", "create", "--name", name}, args...)...)
+	if err != nil {
+		return 0, err
+	}
+
+	if !assert.Regexp(t, `^SSH key [0-9]+ created\n$`, out) {
+		return 0, fmt.Errorf("invalid response: %s", out)
+	}
+
+	id, err := strconv.ParseInt(out[8:len(out)-9], 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	t.Cleanup(func() {
+		_, _ = client.SSHKey.Delete(context.Background(), &hcloud.SSHKey{ID: id})
+	})
+	return id, nil
+}
+
+func generateSSHKey() (string, string, error) {
+	pub, _, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		return "", "", err
+	}
+
+	pubKey, err := ssh.NewPublicKey(pub)
+	if err != nil {
+		return "", "", err
+	}
+
+	fingerprint := ssh.FingerprintLegacyMD5(pubKey)
+	pubKeyBytes := ssh.MarshalAuthorizedKey(pubKey)
+	return string(pubKeyBytes), fingerprint, nil
+}
```

Comment:
We have some util to do this: https://pkg.go.dev/github.com/hetznercloud/hcloud-go/v2@v2.17.1/hcloud/exp/kit/sshutil#GenerateKeyPair